### ISSUE TITLE
[Feature]: Make `outlines` dependency optional

### ISSUE THREAD

**Issue by saattrupdan** [vllm-project/vllm#3794]
### 🚀 The feature, motivation and pitch

I'm using a newer version of `outlines` than v0.0.34, and my application needs the fixes implemented in newer versions of that package. It would be great if `vllm` could make its structured generation dependencies optional, as I'm not using the internal `vllm` structured generation, but rather using the logits processors from `outlines` directly.

### Additional context

This prevents me from updating to `vllm==0.4.0`.

**Comment by simon-mo** [comment]
Alternatively we can relax the outlines version to not to pin it or set it `>=v0.0.34`. That will be an even easier fix at the moment :D. 

Once we have multiple guided decoding backends I think the packages can be optional.

**Comment by saattrupdan** [comment]
> Alternatively we can relax the outlines version to not to pin it or set it `>=v0.0.34`. That will be an even easier fix at the moment :D. 

That could work! However, note that some of the important from outlines (RegexFSM for instance) is currently deprecated and will be removed in June, causing ImportErrors in vLLM if nothing changes. But it would at least work until then :)

**Comment by simon-mo** [comment]
The update ticket: [#outlines](https://github.com/vllm-project/vllm/issues/3715)

**Comment by saattrupdan** [comment]
@simon-mo An upper bound on `outlines` was re-introduced in https://github.com/vllm-project/vllm/pull/6262, and I am relying on newer `outlines` versions. Since I don't use the `vllm`-`outlines` integration at all (I just use the logits processors from outlines), I'd raise my initial question again: Can we please make the `outlines` dependency of vLLM optional?

**Comment by mgoin** [comment]
We will support the latest versions of outlines with this PR https://github.com/vllm-project/vllm/pull/10576
